TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Infrastructure/Filters/CheckPersonExistsFilter.cs

```diff
@@ -38,22 +41,30 @@ public async Task OnResourceExecutionAsync(ResourceExecutingContext context, Res
                     Contact.Fields.dfeta_StatedLastName,
                     Contact.Fields.dfeta_StatedMiddleName,
                     Contact.Fields.dfeta_QTSDate)));
+
         if (person is null)
         {
             context.Result = new NotFoundResult();
             return;
         }
-        else
+        else if (_requireQts && person.Contact.dfeta_QTSDate is null)
         {
-            if (_requireQts && person.Contact.dfeta_QTSDate is null)
-            {
-                context.Result = new BadRequestResult();
-                return;
-            }
+            context.Result = new BadRequestResult();
+            return;
         }
 
-        context.HttpContext.Items["CurrentPersonDetail"] = person;
+        context.HttpContext.SetCurrentPersonInfoFeature(person);
 
         await next();
     }
 }
+
+public class CheckPersonExistsFilterFactory(bool requireQts = false) : IFilterFactory, IOrderedFilter
```

Comment:
This factory means we can use constructor injection on the filter rather than `HttpContext.Services.GetRequiredService...`